# --- T2-COPYRIGHT-NOTE-BEGIN ---
# T2 SDE: package/*/tcp_wrappers/hotfix-gcc14.patch
# Copyright (C) 2024 The T2 SDE Project
# 
# This Copyright note is generated by scripts/Create-CopyPatch,
# more information can be found in the files COPYING and README.
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License version 2 as used by the T2 SDE.
# --- T2-COPYRIGHT-NOTE-END ---

--- tcp_wrappers_7.6/setenv.c.vanilla	2024-03-25 15:48:25.604000000 +0000
+++ tcp_wrappers_7.6/setenv.c	2024-03-25 15:52:45.339000000 +0000
@@ -10,6 +10,10 @@
   * Author: Wietse Venema, Eindhoven University of Technology, The Netherlands.
   */
 
+#include <stdlib.h>
+#include <string.h>
+#include <stdio.h>
+
 #ifndef lint
 static char sccsid[] = "@(#) setenv.c 1.1 93/03/07 22:47:58";
 #endif
@@ -17,11 +21,11 @@
 /* setenv - update or insert environment (name,value) pair */
 
 int     setenv(name, value, clobber)
-char   *name;
-char   *value;
+const char   *name;
+const char   *value;
 int     clobber;
 {
-    char   *malloc();
+    void   *malloc();
     char   *getenv();
     char   *cp;
 
--- tcp_wrappers_7.6/tcpdmatch.c.vanilla	2024-03-25 15:59:43.842000000 +0000
+++ tcp_wrappers_7.6/tcpdmatch.c	2024-03-25 16:00:14.073000000 +0000
@@ -30,6 +30,7 @@
 #include <syslog.h>
 #include <setjmp.h>
 #include <string.h>
+#include <stdlib.h>
 
 extern void exit();
 extern int optind;

